pipeline/filters/lua.md

```diff
@@ -436,3 +436,131 @@ pipeline:
 #### Output
 
 In the output only the messages with response code 0 or greater than 399 are shown.
+
+
+### Timeformat Conversion
+
+In this example, we want to convert specific type of datetime format of a field to `utc ISO 8601` format.
+
+#### Lua script
+
+Script `custom_datetime_format.lua`
+
+```lua
+function convert_to_utc(tag, timestamp, record)
+    local date_time = record["pub_date"]
+    local new_record = record
+    if date_time then
+        if string.find(date_time, ",") then
+            local pattern = "(%a+, %d+ %a+ %d+ %d+:%d+:%d+) ([+-]%d%d%d%d)"
+            local date_part, zone_part = date_time:match(pattern)
+
+            if date_part and zone_part then
+                local command = string.format("date -u -d '%s %s' +%%Y-%%m-%%dT%%H:%%M:%%SZ", date_part, zone_part)
+                local handle = io.popen(command)
+                local result = handle:read("*a")
+                handle:close()
+                new_record["pub_date"] = result:match("%S+")
+            end
+        end
+    end
+    return 1, timestamp, new_record
+end
+```
+
+#### Configuration
+
+Configuration to get a json key with datetime and convert it to another format.
+
+{% tabs %}
+{% tab title="fluent-bit.conf" %}
+```ini
+[INPUT]
+    Name    dummy
```

Comment:
Can we rename this from "dummy" to something less pejorative? Like "placeholder" or "test" or something? And if we can, we'd need to do a search and replace in this to make that change, as it's used in code examples and at least one text line that follows.

Reply:
@cnorris-cs That `Name dummy` is actually another pipeline step from this: [https://docs.fluentbit.io/manual/pipeline/inputs/dummy](https://docs.fluentbit.io/manual/pipeline/inputs/dummy)

Reply:
Yeah this comes up often @cnorris-cs but the actual input plugin is called `dummy` so is a required name for the configuration.